scripts/travis/cleanup.js

```diff
@@ -1,22 +1,38 @@
 /*
- * Delete the artefact from the testing db
+ * Delete old artefacts from the testing db
  */
-
 const { UPLOAD_URL, TRAVIS_BUILD_NUMBER } = process.env;
+const BUILDS_TO_KEEP = 50;
+const END_BUILD_TO_DELETE = TRAVIS_BUILD_NUMBER - BUILDS_TO_KEEP;
+const END_KEY = `medic:medic:test-${END_BUILD_TO_DELETE}`;
+const MAX_BUILDS_TO_DELETE = 50;
 const PouchDB = require('pouchdb-core');
 PouchDB.plugin(require('pouchdb-adapter-http'));
 
 const db = new PouchDB(`${UPLOAD_URL}/_couch/builds_testing`);
-const docId = `medic:medic:test-${TRAVIS_BUILD_NUMBER}`;
 
 const get = () => {
-  console.log(`Getting "${docId}"...`);
-  return db.get(docId);
+  console.log(`Getting builds...`);
+  return db.allDocs({ endkey: END_KEY, limit: MAX_BUILDS_TO_DELETE });
+};
+
+const markDeleted = response => {
+  return response.rows.map(row => {
+    return {
+      _id: row.id,
+      _rev: row.value.rev,
+      _deleted: true
+    };
+  });
 };
 
-const remove = doc => {
-  console.log(`Deleting "${docId}"...`);
-  return db.remove(doc);
+const remove = docs => {
+  if (docs.length) {
+    console.log(`Deleting the oldest ${docs.length} builds...`);
+    return db.bulkDocs(docs);
+  }
 };
 
-get().then(remove);
+get()
+  .then(markDeleted)
+  .then(remove);
```

Comment:
Do we want to consider cleanup here? I guess it depends on how frequently it runs. If it runs every time then presumably we've configured CouchDB to automatically clean up occasionally and it's fine.

Reply:
I've added a db compaction at the end. View cleanup shouldn't be needed because there are no views other than _all_docs which I presume doesn't need cleaned...?